### PR TITLE
Respect original property ordering.

### PR DESCRIPTION
**Pull request by trecloux**
This pull request is following PR #543 from my colleague @gwallet .
We finally found a way to respect the original property ordering.

This pull request includes a unit test.

Thanks, Thomas.

### CODE REVIEW

**Comment by dilipkrish** [comment]
@trecloux thanks for the fix! Still trying to understand how filtering the method with a check on the property name manages to sort it correctly. :confused: 

Before I pull it in, I just want to make sure we don't have a gap in the existing test cases that it may regress.

File: swagger-models/src/test/groovy/com/mangofactory/swagger/models/property/bean/BeanModelPropertyProviderSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    given:
+      Class typeToTest = complexType()
+      def typeResolver = new TypeResolver()
+      ResolvedType resolvedType = typeResolver.resolve(typeToTest)
+      ObjectMapper mapper = new ObjectMapper()
+
+      def beanModelPropertyProvider = new BeanModelPropertyProvider(new AccessorsProvider(typeResolver), typeResolver,
+            new AlternateTypeProvider(), new ObjectMapperBeanPropertyNamingStrategy(mapper))
+      beanModelPropertyProvider.objectMapper = mapper
+      def properties = beanModelPropertyProvider.propertiesForSerialization(resolvedType)
```
Comment (dilipkrish) [inline swagger-models/src/test/groovy/com/mangofactory/swagger/models/property/bean/BeanModelPropertyProviderSpec.groovy]: speficifically, does the following also work, for weird edge cases in the way jackson allows you to configure beans! :)

```java
def properties = beanModelPropertyProvider.propertiesForDeserialization(resolvedType)
```

**Comment by dilipkrish** [comment]
Thanks @trecloux! :+1:
